### DISCUSSION TITLE
Migration Path to Next.JS 13?

### DISCUSSION THREAD

**Discussion by anthonyp-usafacts** [jaredpalmer/razzle#1900] [General]
@jaredpalmer, I appreciate the work and maintenance over the years of this project.

I'm hoping now that you're with Vercel, and after yesterday's big news regarding Turbo and Next.js, that you would consider putting together a plan for those here who are heavily invested in the Razzle/After paradigm to migrate over to this new world.

This would also present an official deprecation plan for Razzle and After.js.

I know you have *a lot* going on. Appreciate your thoughts here.

**Comment by fivethreeo** [comment]
https://github.com/elzzad/dazzle - is a work in progress, jared seems to have abandoned his earlier projects

**Reply by anthonyp-usafacts** [reply]
Hey @fivethreeo I've seen Dazzle but is that going to be a drop-in replacement for Razzle and After.js which helps us get to the Next/Turbo future? Staying with Webpack doesn't make much sense now does it?

If Dazzle has plans, at least documented wish-to-do, to migrate off Webpack to Turbo someday, I would consider that the next step for people on Razzle today.

**Comment by fivethreeo** [comment]
After.js will not be a thing going forward, I am thinking fetch as you render with streaming ssr. Will be making examples. React also will have more support for this with the new use hook. Not quite a drop in, but migration should be easy.

**Reply by anthonyp-usafacts** [reply]
The migration path will be difficult because of integration of Razzle with things like After and Material UI which if you're back on major versions from React, is a huge, huge change set. This is why I pitched this question to Jared because his new role should be very mindful of a huge gap between most of React users and this new tech he's supporting.
